testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -685,6 +691,11 @@ impl BlockMinerThread {
         )
         .expect("FATAL: could not open sortition DB");
 
+        if self.config.get_node_config(false).mock_mining {
```

Comment:
This is a nit and was already in place with the mining_key check before so don't hold up the PR for it, but maybe we should move these config checks to the very top of the function. Will this potentially prevent unnecessary deadlocks for the chainstate and sortition db?

Reply:
good point. May as well move them up.

Reply:
Moved in 5bedd6e43.